### PR TITLE
feat: enable new curriculum

### PR DESCRIPTION
**Pull request by naomi-lgbt**
This PR will ultimately allow for the launch of the v7 (v8?) Responsive Web Design.

Currently this PR does:
- Adds an `isLegacy` flag for blocks in the `meta.json` files
- Uses the `isLegacy` flag to determine blocks that are no longer a part of the core curriculum, and separates them on the superblock view
- Adds some text explaining that the `isLegacy` sections are legacy, but still available for campers to further their learning.
- Reorders the 

This PR still needs to:
- Pull in the remaining 3 projects when complete, and ensure the order is correct.
- Determine why the `Responsive Web Design` block has disappeared from the map.
- ~Fix the donation flow: it should report partial progress as you complete projects and only request donations when you're done.~
- Send the user to learn/responsive-web-design/#claim-cert-block when all 5 projects are done, not to learn/responsive-web-design/applied-accessibility/know-when-alt-text-should-be-left-blank

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
@nhcarrigan I put everything in the order determined by the sheet.  There shouldn't be any need to re-order once the final practice projects are merged, because I left gaps for them.

This isn't *done*, since I broken the donation flow, but feel free to rebase and I'll tackle that once you're done.

**Comment by nhcarrigan** [comment]
image

Well, I gave it a rebase, but the thing I thought would be fixed isn't. 😦

**Comment by ojeytonwilliams** [comment]
Ah, presumably it's hidden because of `isLegacy`.  I can take look tomorrow, if you like.

**Comment by nhcarrigan** [comment]
@ojeytonwilliams I found the bug.

The GraphQL query was looking for block `2`, which in RWD is the marker project we don't have yet. I've updated the GraphQL query to look for block `1` (which is the second block in a 0-based index approach, so properly reflects the comment).

The Front End Libraries now disappears because the first block starts at `2`. I know there was a PR to fix the challenge orders, but I believe that became decayed. I'll start a new one separate from this. 🙂

**Comment by ojeytonwilliams** [comment]
Thanks, Nick, the fix is working. I suspect we should either use 1-based order for everything and just pick the first challenge in the first block.  Or, if that doesn't work, explain why we're doing it in a weird way!

File: client/src/templates/Challenges/classic/desktop-layout.tsx
Comment (nhcarrigan) [inline client/src/templates/Challenges/classic/desktop-layout.tsx:112]: This might be a stray change from the rebase.

**Comment by ojeytonwilliams** [comment]
@nhcarrigan when you rebase, could you keep the last commit separate?  It messes up the order and I haven't been able to figure out what's going wrong.

**Comment by ojeytonwilliams** [comment]
The issue is the duplicated `challenge.id`s - since the 'legacy' projects really are just the same challenges as the 'new' projects, they should have the same mongo id (`challenge.id`).  However, that id is used by Gatsby and must be unique.

**Comment by ojeytonwilliams** [comment]
We now have two superBlocks /learn/responsive-web-design and the imaginatively titled /learn/responsive-web-design-two

I went with renaming the new stuff simply because I had to pick something.  It's relatively straightforward to change the names of those two, but it might involve a few hacks when we finally get to the localised clients.

Since the responsive-web-design certification can now be completed in two superBlocks, there's no longer a 1-1 map between superBlocks and certifications.  This had surprisingly few consequences, but it did mean we needed a way of knowing which certification a given superBlock is connected to - hence the `certification` property for the certification.ymls and, for the client, the superBlock markdown files also got a new property.

The remaining issues that I'm aware of:

- /settings has both RWD and RWD2
- on /settings the RWD2 section has the RWD links

After fixing those, the next steps will be:

- Use Nick's flag to control if the curriculum builds the new superBlock *and* the superBlock order.
- Flip the flag in the pipelines when we want to deploy

This will mean that we can merge this into main and nothing will happen except that we can stop rebasing this.  Hopefully.

File: client/i18n/locales/english/intro.json
Comment (ShaunSHamilton) [inline client/i18n/locales/english/intro.json:87]: I understand the duplicate certification project intros, but seeing as these "do not" (should not) appear within `responsive-web-design`, can these not be removed?

**Comment by moT01** [comment]
It’s looking good 👍 Some of these aren’t necessarily related to this specific PR - but here’s what I got…

I mentioned this already, but I would definitely go with giving the legacy a stuff a new name, and keep “Responsive Web Design” as the name for the first superblock.

I’m not sure, but I think we will eventually phase the cert projects out of the legacy RWD superblock - and just turn it into a "Legacy RWD Challenges" or something.

On the settings page, I don’t think we need RWD and RWD2 - we aren’t creating a new cert. We still only have the RWD Cert. If they complete the projects in the RWD or RWD2 superblock, it doesn’t matter. They get the RWD Cert. So, I think I would vote to only have the new RWD on the settings page. The project links there would go to the new RWD, and if someone wants to complete them in the other section, that works too.

There seemed to be a little bit of goofiness around the cert claiming/viewing. But on a second try, it seems to be working good… I think. Only thing is the claim cert button doesn’t change to show cert on the superblock page after I’ve claimed it. Shows the same on the settings page.

The rest of these are pretty minor:
Also on the setting page, the order of the RWD projects isn’t aligned with the new order they appear in the curriculum.

Could maybe use a “Legacy Challenges” or something header on the map between the coding interview prep and the RWD at the bottom

We could maybe get rid of the progress bar in the completion modal. We don’t show the modal when finishing a step, and it doesn’t make all that much sense. It says “Build a Tribute Page Project - 40% Complete”. Perhaps, it we could change it to “Responsive Web Design - 40% Complete”

**Comment by ojeytonwilliams** [comment]
> I mentioned this already, but I would definitely go with giving the legacy a stuff a new name, and keep “Responsive Web Design” as the name for the first superblock.

That sounds good.  I think we should use a new slug, but keep the name unchanged.  `responsive-web-design-v2` for the slug, maybe?  

> On the settings page, I don’t think we need RWD and RWD2 - we aren’t creating a new cert. We still only have the RWD Cert. If they complete the projects in the RWD or RWD2 superblock, it doesn’t matter. They get the RWD Cert. So, I think I would vote to only have the new RWD on the settings page. The project links there would go to the new RWD, and if someone wants to complete them in the other section, that works too.

Agreed, definitely for the first iteration.

> Could maybe use a “Legacy Challenges” or something header on the map between the coding interview prep and the RWD at the bottom

Sure, but let's do that in a follow up PR.

> We could maybe get rid of the progress bar in the completion modal. We don’t show the modal when finishing a step, and it doesn’t make all that much sense. It says “Build a Tribute Page Project - 40% Complete”. Perhaps, it we could change it to “Responsive Web Design - 40% Complete”

Same.  It's reasonably straightforward (the cert title can be fetched by graphQL and is translated), but I'd rather do that after this PR is in.

**Comment by ojeytonwilliams** [comment]
I think everything has been addressed now, including the 'Show/Claim' cert button. If `showNewCurriculum` and `showUpcomingChanges` are both false, this PR should do almost nothing.  The only change I know of is that the responsive web design projects are ordered differently on /settings (not in the map, though),

If we flip those flags, the curriculum will build an additional superBlock `responsive-web-design-v2`, and that will appear first in the map.  The old curriculum will still live at `responsive-web-design`, but will be pushed to the end of the map.

**Comment by ShaunSHamilton** [comment]
For the most part, this is LGTM 👍 

The main change I suggest is the change of the url to `responsive-web-design-2021` (or 2022 considering that will be the final final release).

Though, that does not have to be in this PR.

**Comment by nhcarrigan** [comment]
> The main change I suggest is the change of the url to responsive-web-design-2021 (or 2022 considering that will be the final final release).

I think we should land on the final URL in this PR - because the URL is determined by changing the folder name, changing it again in another PR just creates additional noise/mess.

**Review by moT01 — DISMISSED** [review]
Nicely done @nhcarrigan & @ojeytonwilliams 👍 LGTM 🎉 I would fine merging this - main thing is that with both the env variables set to false, everything looks and works the same as production. So we should be good.

There is one issue I think I'm seeing - should be minor. The meta for the practice projects all still have the `isUpcomingChange` set to `true`. With `show_new=true` and `show_upcoming=false`, the first RWD superblock doesn't show up on the map. You can get there by entering the URL. Is there a bug there? Maybe a test that doesn't exist, or isn't working as expected? Adding `false` to all the meta files looks like it works as intended. I vote to change on another PR.

I don't think we should hold up the PR, but we should run the path thing my Mrugesh - we all kind of concluded that this is the way to go. So I say we go with it. Nothing will be live yet.

Here's my other findings that we can add/fix on another PR if they're needed:
- +1 on the v2 rename - I would go with `responsive-web-design-2022` - or, I dunno - maybe not. In 2025 will the date make it feel outdated?

- rename the bottom RWD button so it isn't the same as the top one. Or do something to distinguish it anyway

- add intros for the new project blocks

- add a comment in the codebase somewhere that says `TODO: search the codebase for "TODO", and do some of them`

- Naming convention for the new projects - I will mention this in chat


Yea, pretty much no real issues here - looks good

**Comment by nhcarrigan** [comment]
> The meta for the practice projects all still have the isUpcomingChange set to true.

I thought I removed that. Will take another look.

**Comment by nhcarrigan** [comment]
Okay I have confirmed that:

- With `SHOW_UPCOMING_CHANGES=false` and `SHOW_NEW_CURRICULUM=true`, the new RDW shows now.
- With the language set to `espanol` and both flags set to `false`, the build does not fail (it does not look for the new curriculum files which are currently not present).

**Review by moT01 — DISMISSED** [review]
LGTM 🎉

**Review by ShaunSHamilton — DISMISSED** [review]
LFTM 👍 

I am not a fan of the `superblockToNewOrder` existence, but assume this will replace the old order, once final final release.

**Comment by ahmadabdolsaheb** [comment]
Great job working on this. Here are a few points that we could consider in the upcoming pull-requests.

Preferably we would want to have one RWD on the map (might be a personal preference, but sounds more intuitive).

Additionally, we would need a way to navigate between the two versions of the curriculum. An alert section with a button should do)

Since the new curriculum, editor, etc has not been tested by users yet and a large segment of our user are going to be affected with this change, it would be great to make this feature initially opt-in every-time until we make sure it is solid and vital metrics are not affected.

Once we are more confident, we could reverse that route, have the users go to the new curriculum from the map, click on a button and  see the old changes. 

It would be great if we could do usability testing before rolling out this feature. Nothing fancy. Just watching a few users navigate the UI and do a couple of challenges would do.

**Comment by ShaunSHamilton** [comment]
Where are we on landing this? Mrugesh has suggested the `/learn/2022` paths, but I wonder if you all want to get this in, and do that (and a few other things) in another PR?

**Comment by moT01** [comment]
I vote to get it in. Otherwise, this PR just keeps getting bigger and more complex.

**Comment by ahmadabdolsaheb** [comment]
Here are some additional UX considerations to keep in mind for the future development:

- The block completion donation modal would only pop up after the final project is completed (expected) and have the name of the last project in the title. Might make sense to pop the donation modal at the end of each project.

- We would need to have a way to visually tell the cert projects a part from the rest of the cards, or group related cards together based on their related cert project.

- It might take some getting used to, but redirecting to the next practice project right after submitting a cert project seems confusing. Taking the related part of the map to see next project and their progress might be a better experience.

**Comment by ojeytonwilliams** [comment]
@ahmadabdolsaheb could you edit in your recommendations to https://github.com/freeCodeCamp/freeCodeCamp/issues/43668 just to make sure we keep everything in one place?

**Review by ShaunSHamilton — APPROVED** [review]
Pre-emptively thumbs-upping 👍

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**
